### PR TITLE
react carousel component - Logan Lee

### PR DESCRIPTION
**Pull request by loganlee-lth**
Kapture 2023-07-13 at 13 33 22

### CODE REVIEW

File: react-carousel-component/my-app/src/Carousel.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import Indicators from './Indicators';
+import './Carousel.css';
+
+export default function Carousel({ images }) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  useEffect(() => {
+    const interval = setInterval(handleClickNext, 3000);
+    return () => clearInterval(interval);
+  });
```
Comment (RobertGardner) [inline react-carousel-component/my-app/src/Carousel.js:14]: Where is your dependency array? Without it, this effect will run on _every_ render, which means that on every render the clean up function will get called and the interval will be cleared, then you'll start it again. So the 3 second interval will start over on each render. That is not what you want!

File: react-carousel-component/my-app/src/Carousel.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  function handleClickPrev() {
+    const index = (currentIndex - 1 + images.length) % images.length;
+    setCurrentIndex(index);
+  }
+
+  function handleClickNext() {
```
Comment (RobertGardner) [inline react-carousel-component/my-app/src/Carousel.js:21]: You should find, when you fix the dependency in useEffect, that this now causes the effect to run on every render. Do you remember how to fix that?

**Review by RobertGardner — CHANGES_REQUESTED** [review]
Fix the problem with useEffect.

File: react-carousel-component/my-app/src/Carousel.js (hunk 1 above)
Comment (loganlee-lth) [reply react-carousel-component/my-app/src/Carousel.js]: In line 11, useEffect includes function defined outside its scope, so I added handleClickNext in the dependency array.

**Review by RobertGardner — APPROVED** [review]

**Pull request merged**